### ISSUE TITLE
Debugging Issue: Pydev "couldn't send frame 501"/No connection (command: 501) & PathMapper: "failed to map"

### ISSUE THREAD

**Issue by irishgordo** [aws/aws-toolkit-jetbrains#1725]
**Describe the bug**
Unable to debug, step through code lines, pydev fails to get frame 501 or something, also Path Mapping fails.

**Note**
"renamed the project that I was working on to 'project'...just not to put sensitive info out there haha"

**To reproduce**
Run Debug on configuration:

```
Invoking src.main.handler (python3.8)

Fetching lambci/lambda:python3.8 Docker container image......
Mounting /Users/mike.russell/Projects/project/.aws-sam/build/Lambda as /var/task:ro,delegated inside runtime container
Connected to pydev debugger (build 201.6668.115)
pydev debugger: warning: trying to add breakpoint to file that does not exist: /Users/mike.russell/Projects/project/src/main.py (will have no effect)
START RequestId: 9010117e-6b64-103a-9cfd-0ed46a3e60cb Version: $LATEST
{'Records': [{'messageId': '19dd0b57-b21e-4ac1-bd88-01bbb068cb78', 'receiptHandle': 'MessageReceiptHandle', 'body': 'Hello from SQS!', 'attributes': {'ApproximateReceiveCount': '1', 'SentTimestamp': '1523232000000', 'SenderId': '123456789012', 'ApproximateFirstReceiveTimestamp': '1523232000001'}, 'messageAttributes': {}, 'md5OfBody': '7b270e59b47ff90a553787216d55d91d', 'eventSource': 'aws:sqs', 'eventSourceARN': 'arn:{partition}:sqs:{region}:123456789012:MyQueue', 'awsRegion': '{region}'}]}
<class 'dict'>
<bootstrap.LambdaContext object at 0x7fd40c557d90>
<class 'bootstrap.LambdaContext'>
END RequestId: 9010117e-6b64-103a-9cfd-0ed46a3e60cb
REPORT RequestId: 9010117e-6b64-103a-9cfd-0ed46a3e60cb	Init Duration: 1186.45 ms	Duration: 4.04 ms	Billed Duration: 100 ms	Memory Size: 128 MB	Max Memory Used: 36 MB	

"Hello World"

```

**Expected behavior**
To be able to step through calls

**Screenshots**
pydev debugger: warning: trying to add breakpoint to file that does not exist: /Users/mike.russell/Projects/project/src/main.py (will have no effect)
![Screen Shot 2020-04-23 at 10 47 21 AM](https://user-images.githubusercontent.com/5370752/80132159-20a84d00-8550-11ea-9561-7aa6c74c6eb6.png)

**Your Environment**
PyCharm 2020.1 (Professional Edition)
Build #PY-201.6668.115, built on April 7, 2020
Licensed to Michael Russell
Subscription is active until March 9, 2021
Runtime version: 11.0.6+8-b765.25 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.4
GC: ParNew, ConcurrentMarkSweep
Memory: 1979M
Cores: 12
Non-Bundled Plugins: IdeaVIM, com.andrey4623.rainbowcsv, io.protostuff.protostuff-jetbrains-plugin, net.seesharpsoft.intellij.plugins.csv, aws.toolkit, com.jetbrains.intellij.datalore, com.leinardi.pycharm.pylint, ru.meanmail.plugin.requirements
Python environment: Virtual Env, "venv" in ".env" directory in root of project, Python 3.8

**Additional context**
**Debug Logs**
```
2020-04-23 11:00:54,601 [ 237580]   INFO - .services.lambda.LambdaBuilder - /usr/local/bin/sam build Lambda --template /Users/mike.russell/Projects/project/template.yaml --build-dir /Users/mike.russell/Projects/project/.aws-sam/build --use-container 
2020-04-23 11:00:54,627 [ 237606]   INFO - idation.LambdaHandlerValidator - Evaluating LambdaEntry(project=Project (name=project, containerState=ACTIVE, componentStore=/Users/mike.russell/Projects/project) , runtime=python3.8, handler=src.main.handler) 
2020-04-23 11:00:55,564 [ 238543]   INFO - .services.lambda.LambdaBuilder - Starting Build inside a container 
2020-04-23 11:00:55,990 [ 238969]   INFO - .services.lambda.LambdaBuilder - Building resource 'Lambda' 
2020-04-23 11:00:57,251 [ 240230]   INFO - .services.lambda.LambdaBuilder -  
2020-04-23 11:00:57,252 [ 240231]   INFO - .services.lambda.LambdaBuilder - Fetching lambci/lambda:build-python3.8 Docker container image...... 
2020-04-23 11:00:57,252 [ 240231]   INFO - .services.lambda.LambdaBuilder - Mounting /Users/mike.russell/Projects/project as /tmp/samcli/source:ro,delegated inside runtime container 
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder -  
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder - Build Succeeded 
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder -  
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder - Built Artifacts  : Users/mike.russell/Projects/project/.aws-sam/build 
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder - Built Template   : Users/mike.russell/Projects/project/.aws-sam/build/template.yaml 
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder -  
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder - Commands you can use next 
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder - ========================= 
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder - [*] Invoke Function: sam local invoke -t Users/mike.russell/Projects/project/.aws-sam/build/template.yaml 
2020-04-23 11:01:04,954 [ 247933]   INFO - .services.lambda.LambdaBuilder - [*] Deploy: sam deploy --guided --template-file Users/mike.russell/Projects/project/.aws-sam/build/template.yaml 
2020-04-23 11:01:04,955 [ 247934]   INFO - .services.lambda.LambdaBuilder -  
2020-04-23 11:01:05,775 [ 248754]   INFO - .services.lambda.LambdaBuilder - Running PythonPipBuilder:ResolveDependencies 
2020-04-23 11:01:05,776 [ 248755]   INFO - .services.lambda.LambdaBuilder - Running PythonPipBuilder:CopySource 
2020-04-23 11:01:06,098 [ 249077]  DEBUG - .jetbrains.services.PathMapper - Failed to map /Users/mike.russell/Projects/project/src/main.py to remote file system: [PathMapping(localRoot='/Users/mike.russell/Projects/project/<mapping:2dfac4cb>/', remoteRoot='/var/task/'), PathMapping(localRoot='/Users/mike.russell/Projects/project/.aws-sam/build/Lambda/', remoteRoot='/var/task/'), PathMapping(localRoot='/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/', remoteRoot='/tmp/lambci_debug_files/')] 
2020-04-23 11:01:06,109 [ 249088]   INFO - .idea.vim.command.CommandState - Push new mode state: COMMAND:NONE 
2020-04-23 11:01:06,109 [ 249088]   INFO - .idea.vim.command.CommandState - Push new mode state: INSERT:NONE 
2020-04-23 11:01:06,276 [ 249255]   INFO - idation.LambdaHandlerValidator - Evaluating LambdaEntry(project=Project (name=project, containerState=ACTIVE, componentStore=/Users/mike.russell/Projects/project) , runtime=python3.8, handler=src.main.handler) 
2020-04-23 11:01:09,738 [ 252717]   INFO - idation.LambdaHandlerValidator - Evaluating LambdaEntry(project=Project (name=project, containerState=ACTIVE, componentStore=/Users/mike.russell/Projects/project) , runtime=python3.8, handler=src.main.handler) 
2020-04-23 11:01:09,883 [ 252862]  ERROR - debugger.pydev.AbstractCommand - No connection (command:  501 ) 
java.lang.Throwable: No connection (command:  501 )
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:146)
	at com.jetbrains.python.debugger.pydev.AbstractCommand.execute(AbstractCommand.java:154)
	at com.jetbrains.python.debugger.pydev.RemoteDebugger.handshake(RemoteDebugger.java:131)
	at com.jetbrains.python.debugger.pydev.transport.ClientModeDebuggerTransport.waitForConnect(ClientModeDebuggerTransport.java:102)
	at com.jetbrains.python.debugger.pydev.RemoteDebugger.waitForConnect(RemoteDebugger.java:114)
	at com.jetbrains.python.debugger.pydev.ClientModeMultiProcessDebugger.tryToConnectRemoteDebugger(ClientModeMultiProcessDebugger.java:96)
	at com.jetbrains.python.debugger.pydev.RecurrentTaskExecutor.tryToPerformTaskAndHandleSuccess(RecurrentTaskExecutor.kt:174)
	at com.jetbrains.python.debugger.pydev.RecurrentTaskExecutor.access$tryToPerformTaskAndHandleSuccess(RecurrentTaskExecutor.kt:45)
	at com.jetbrains.python.debugger.pydev.RecurrentTaskExecutor$TaskRunnable.run(RecurrentTaskExecutor.kt:100)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
2020-04-23 11:01:09,885 [ 252864]  ERROR - debugger.pydev.AbstractCommand - PyCharm 2020.1  Build #PY-201.6668.115 
2020-04-23 11:01:09,885 [ 252864]  ERROR - debugger.pydev.AbstractCommand - JDK: 11.0.6; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o 
2020-04-23 11:01:09,885 [ 252864]  ERROR - debugger.pydev.AbstractCommand - OS: Mac OS X 
2020-04-23 11:01:09,885 [ 252864]  ERROR - debugger.pydev.AbstractCommand - Last Action: Debug 
```

**Comment by abrooksv** [comment]
We will need debug logs to triage the path mapping error.

Open the debug logs settings and add `software.aws.toolkits`
It can be found under Help->Diagnostic Tools menu.

**Comment by irishgordo** [comment]
Hey @abrooksv 
thanks for the super quick response!
I've really enjoyed this awesome toolkit!

I went ahead and added that in the debug log settings, restarted pycharm, but then when I open up the idea.log, all I see is the same debugging that I had originally posted with just different timestamps.

Aside from this callout to "software.aws.tookits":
```
2020-04-23 13:00:37,165 [   5084]   WARN - Container.ComponentManagerImpl - Do not use constructor injection (requestorClass=software.aws.toolkits.jetbrains.core.region.AwsRegionProvider) 
```

I also tried adding "software.aws.toolkits:trace" and restarting pycharm but I didn't see anything different.

**Comment by abrooksv** [comment]
```
2020-04-23 13:10:18,209 [  53072]  DEBUG - .jetbrains.services.PathMapper - Failed to map /Users/mike.russell/Projects/project/src/main.py to remote file system: [PathMapping(localRoot='/Users/mike.russell/Projects/project/<mapping:2dfac4cb>/', remoteRoot='/var/task/'), PathMapping(localRoot='/Users/mike.russell/Projects/project/.aws-sam/build/Lambda/', remoteRoot='/var/task/'), PathMapping(localRoot='/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/', remoteRoot='/tmp/lambci_debug_files/')]
``` 

is what I was looking for, thanks

**Comment by abrooksv** [comment]
Basically we are failing to translate the local file (main.py) to the remote file system's path so pydevd knows where to put the breakpoint.

**Comment by abrooksv** [comment]
Found the odd string: It is the toString() method of the YAMLMapping of the template. (https://github.com/JetBrains/intellij-community/blob/master/plugins/yaml/src/org/jetbrains/yaml/psi/impl/YAMLMappingImpl.java#L68)

Can you tell me a bit about your template file?

My hunch is its:

```
        val codeLocation = ReadAction.compute<String, Throwable> {
            functions.find { it.logicalName == logicalId }
                ?.codeLocation()
                ?: throw RuntimeConfigurationError(
                    message(
                        "lambda.run_configuration.sam.no_such_function",
                        logicalId,
                        templateLocation
                    )
                )
        }
```

but I am not sure why that code would lead to the error we are seeing

**Comment by irishgordo** [comment]
@abrooksv  man, I'm so sorry to waste your time on this.
I ended up editing the template to be have the:
```
Transform: AWS::Serverless-2016-10-31
```
and tweaking my prior cloudformation and it works beautifully now.
I guess for some reason I got wrapped up in not converting my function from generic cloudformation to severless-sam based cloudformation.

Thanks again for all your support! 
I really appreciate it!
